web/src/components/health/health-helper.ts

```diff
@@ -0,0 +1,340 @@
+import { PrometheusAlert, PrometheusLabels, Rule } from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash';
+import { SilenceMatcher } from '../../api/alert';
+
+export type HealthStats = {
+  global: ByResource;
+  byNamespace: ByResource[];
+  byNode: ByResource[];
+};
+
+export type ByResource = {
+  name: string;
+  critical: SeverityStats;
+  warning: SeverityStats;
+  other: SeverityStats;
+  score: number;
+};
+
+type SeverityStats = {
+  firing: AlertWithRuleName[];
+  pending: AlertWithRuleName[];
+  silenced: AlertWithRuleName[];
+  inactive: string[];
+};
+
+export type AlertWithRuleName = PrometheusAlert & {
+  ruleName: string;
+  ruleID: string;
+  metadata: HealthMetadata;
+};
+
+type RuleWithMetadata = Rule & {
+  metadata: HealthMetadata;
+};
+
+type HealthMetadata = {
+  threshold?: string;
+  thresholdF: number;
+  upperBound: string;
+  upperBoundF: number;
+  unit: string;
+  nodeLabels?: string[];
+  namespaceLabels?: string[];
+  links: { name: string; url: string }[];
+  trafficLinkFilter?: string;
+};
+
+type ScoreDetail = {
+  rawScore: number;
+  weight: number;
+};
+
+const getHealthMetadata = (annotations: PrometheusLabels): HealthMetadata => {
+  const defaultMetadata: HealthMetadata = {
+    thresholdF: 0,
+    upperBound: '100',
+    upperBoundF: 100,
+    unit: '%',
+    links: []
+  };
+  if ('netobserv_io_network_health' in annotations) {
+    const md = (JSON.parse(annotations['netobserv_io_network_health']) as HealthMetadata) || undefined;
+    if (md) {
+      // Setup defaults and derived
+      md.unit = md.unit || defaultMetadata.unit;
+      md.upperBound = md.upperBound || defaultMetadata.upperBound;
+      md.links = md.links || defaultMetadata.links;
+      md.thresholdF = md.threshold ? parseFloat(md.threshold) || 0 : 0;
```

Comment:
```suggestion
      md.thresholdF = md.threshold ? parseFloat(md.threshold) : 0;
```

`parseFloat` is supposed to return a number anyways

Reply:
it can return NaN
Especially as that's read from external resources, potentially user-provided, I prefer to have this guard rail